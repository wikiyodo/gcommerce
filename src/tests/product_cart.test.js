/**
 * Test login modal
 */

import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import ProductWithColorsCard from '../common/cards/product_buy_now';
import { Route, Switch, BrowserRouter as Router, Redirect } from "react-router-dom";
import { shallow } from 'enzyme';


describe('ProductWithColorsCard  ', ()=>{
    
    it('renders without crashing', ()=>{
        const div = document.createElement('div');
        let cart = <Router>
            <ProductWithColorsCard />
        </Router>;
        ReactDOM.render(cart, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it('renders correctly with product', ()=>{
        const div = document.createElement('div');
        let product = <Router><ProductWithColorsCard product={{
            "product_id": 2,
            "name": "Chartres Cathedral",
            "description": "\"The Fur Merchants\". Not all the beautiful stained glass in the great cathedrals depicts saints and angels! Lay aside your furs for the summer and wear this beautiful T-shirt!",
            "price": "16.95",
            "discounted_price": "15.95",
            "thumbnail": "chartres-cathedral-thumbnail.gif"
            }} /></Router>;
        const component = renderer.create(product);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    
});
