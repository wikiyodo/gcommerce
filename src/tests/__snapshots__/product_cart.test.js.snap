// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductWithColorsCard   renders correctly with product 1`] = `
<div
  className="card-01-sx card "
>
  <div
    className="card-body"
  >
    <div
      className="card-img"
    >
      <img
        src="https://backendapi.turing.com/images/products/chartres-cathedral-thumbnail.gif"
      />
    </div>
    <div
      className="card-title"
    >
      Chartres Cathedral
    </div>
    <div
      className="card-fo"
    >
      <a
        className="btn  btn-gr btn-pink-red"
        href="/products/details/2"
        onClick={[Function]}
      >
        Buy now
      </a>
    </div>
  </div>
</div>
`;
